minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -600,6 +602,97 @@ public synchronized void start() throws IOException, InterruptedException {
     if (executor == null) {
       executor = Executors.newSingleThreadExecutor();
     }
+
+    verifyUp();
+
+  }
+
+  private void verifyUp() throws InterruptedException, IOException {
```

Comment:
It might be useful to make this public (and available outside of the impl) so tests can use it as well. I am pretty sure there are a few tests doing this already, maybe this could replace duplicate code.

Reply:
Do you know which tests?

Reply:
I couldn't think of any off the top of my head but I did a quick search of ITs that look for ZK locks and found: RestartIT, ReadWriteIT, ThriftServerBindsBeforeZooKeeperLockIT, TabletStateChangeIteratorIT

Reply:
And this could be a follow on task since its probably a lot of work.